e2e/testsuite/sanitize/messages.go

```diff
@@ -78,6 +94,44 @@ func removeUnknownFields(tag string, msg sdk.Msg) sdk.Msg {
 		if !icaUnorderedChannelFeatureReleases.IsSupported(tag) {
 			msg.Ordering = channeltypes.NONE
 		}
+	case *clienttypes.MsgUpdateClient:
+		if !cometBFTv1Validator.IsSupported(tag) {
+			clientMessage, err := clienttypes.UnpackClientMessage(msg.ClientMessage)
+			if err != nil {
+				panic(err)
+			}
+			header, ok := clientMessage.(*ibctm.Header)
+			if !ok {
+				return msg
+			}
+
+			replaceCometBFTValidatorV1(header.ValidatorSet.Proposer)
+			for _, validator := range header.ValidatorSet.Validators {
+				replaceCometBFTValidatorV1(validator)
+			}
+
+			replaceCometBFTValidatorV1(header.TrustedValidators.Proposer)
+			for _, validator := range header.TrustedValidators.Validators {
+				replaceCometBFTValidatorV1(validator)
+			}
+
+			// repack the client message
+			clientMessageAny, err := clienttypes.PackClientMessage(header)
+			if err != nil {
+				panic(err)
+			}
+			msg.ClientMessage = clientMessageAny
+		}
 	}
 	return msg
 }
+
+func replaceCometBFTValidatorV1(validator *cmtproto.Validator) {
```

Comment:
nit: maybe mutate or convert instead of replace. since i expected this to be returning a new type rather than mutating in-place

Reply:
Good point! Done!